charts/karpenter/values.yaml

```diff
@@ -6,3 +6,6 @@ controller:
   tolerations: []
   affinity: {}
   image: "public.ecr.aws/karpenter/controller:v0.2.3@sha256:ad4332ddce1afb1080b269eeee70592c29e3e34ecd9f9f4dabe4102c415817e9"
+
+aws:
```

Comment:
Do you mind implementing this using a env map? This will allow you to wire this in without AWS specific changes.
```
    {{- range .Values.controller.environment }}
    - name: {{ .name }}
      value: {{ .value }}
    {{- end }} 
```

Reply:
I agree it would be useful to allow passing arbitrary env vars to the controller. I'm thinking this could be done similarly to how it's dealt with in cluster autoscaler ([ref](https://github.com/helm/charts/blob/fd80db1fed08f4e4b9555d739a10f0242fec34bf/stable/cluster-autoscaler/templates/deployment.yaml#L145)), so a user could end up with something like:

```
aws:
  region: eu-west-1

controller:
  extraEnv:
    MY_LOGCONFIG: my_value
```

The upside of this approach is the clear way of specifying certain well-known options, while the downside is that a user could end up with duplicate definitions of env vars, which would be accepted by k8s, but not reflected properly inside the container.

What's your thoughts?

Reply:
I'd prefer simply the above:
```
    {{- range .Values.controller.environment }}
    - name: {{ .name }}
      value: {{ .value }}
    {{- end }} 
```
The cluster autoscaler suffers from complexity bloat, which is one of the reasons I'm trying to draw a stronger line on our helm chart. I don't think it results in significantly more compelx helm commands. WDYT?

Reply:
Given that your suggested approach provides better flexibility, let's go with that. It is however a bit quirky to set those values from the helm command line given the [limitations of](https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set) `--set`. To achieve the desired result I end up adding `--set "controller.environment[0].name=AWS_REGION,controller.environment[0].value=eu-west-1"`, as opposed to `--set aws.region=eu-west-1` to my commands.

If the opinion changes, it's easier to add specific values rather than remove them.